### PR TITLE
Add helm chart for AKO Operator

### PR DESCRIPTION
**Pull request by hemantavi**
This also fixes a permission issue for clusterrole and clusterrolebinding for the ako operator.

### CODE REVIEW

**Review by akshayhavile — COMMENTED** [review]
Reviewed code from namespace sync functionality perspective. LGTM. One minor comment has been mentioned.

File: ako-operator/helm/ako-operator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  pspEnable: false
+
+
+avicredentials:
+  username: "admin"
+  password: "avi123"
+  certificateAuthorityData:
+
+
+service:
```
Comment (parimanur) [inline ako-operator/helm/ako-operator/values.yaml:92]: can we delete this section?

File: ako-operator/helm/ako-operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+L4Settings:
+  advancedL4: "false" # Use this knob to control the settings for the services API usage. Default to not using services APIs: https://github.com/kubernetes-sigs/service-apis
+  defaultDomain: "" # If multiple sub-domains are configured in the cloud, use this knob to set the default sub-domain to use for L4 VSes.
+
+### This section outlines settings on the Avi controller that affects AKO's functionality.
+ControllerSettings:
+  serviceEngineGroupName: "Default-Group" # Name of the ServiceEngine Group.
+  controllerVersion: "20.1.1" # The controller API version
+  cloudName: "Default-Cloud" # The configured cloud name on the Avi controller.
+  controllerIP: "10.79.111.29"
```
Comment (monotosh-avi) [inline ako-operator/helm/ako-operator/values.yaml:65]: I think we can keep it empty

File: ako-operator/helm/ako-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+### This section outlines all the knobs  used to control Layer 4 loadbalancing settings in AKO.
+L4Settings:
+  advancedL4: "false" # Use this knob to control the settings for the services API usage. Default to not using services APIs: https://github.com/kubernetes-sigs/service-apis
+  defaultDomain: "" # If multiple sub-domains are configured in the cloud, use this knob to set the default sub-domain to use for L4 VSes.
+
+### This section outlines settings on the Avi controller that affects AKO's functionality.
+ControllerSettings:
+  serviceEngineGroupName: "Default-Group" # Name of the ServiceEngine Group.
+  controllerVersion: "20.1.1" # The controller API version
```
Comment (monotosh-avi) [inline ako-operator/helm/ako-operator/values.yaml:63]: I think we support controller version>=20.1.2+ and >=18.2.10

File: ako-operator/helm/ako-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  ## Node Network details are not needed when in nodeport mode / static routes are disabled / non vcenter clouds.
+  nodeNetworkList: []
+  # nodeNetworkList:
+  #   - networkName: "network-name"
+  #     cidrs:
+  #       - 10.0.0.1/24
+  #       - 11.0.0.1/24
+  subnetIP: "10.10.0.0" # Subnet IP of the vip network
+  subnetPrefix: "16" # Subnet Prefix of the vip network
+  networkName: "vcd-ns" # Network Name of the vip network
```
Comment (monotosh-avi) [inline ako-operator/helm/ako-operator/values.yaml:44]: We can keep it empty

**Review by monotosh-avi — COMMENTED** [review]
Looks good. A couple of minor comments. 
Also do we need NOTES.txt ? It is empty.

File: ako-operator/helm/ako-operator/crds/ako.vmware.com_akoconfigs.yaml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+                  description: LogLevel defines the log level to be used by the AKO
+                    controller
+                  enum:
+                  - INFO
+                  - DEBUG
+                  - WARN
+                  - ERROR
+                  type: string
+                namespaceSelector:
+                  description: Namespace selector specifies the namespace labels from
```
[the remark this answers is not in the document]
Comment (hemantavi) [reply ako-operator/helm/ako-operator/crds/ako.vmware.com_akoconfigs.yaml]: Done, thanks.

File: ako-operator/helm/ako-operator/values.yaml (hunk 4 above)
Comment (hemantavi) [reply ako-operator/helm/ako-operator/values.yaml]: Yeah, this and all the other fields in `NetworkSettings`.

File: ako-operator/helm/ako-operator/values.yaml (hunk 3 above)
Comment (hemantavi) [reply ako-operator/helm/ako-operator/values.yaml]: Yeah, changed.

File: ako-operator/helm/ako-operator/values.yaml (hunk 2 above)
Comment (hemantavi) [reply ako-operator/helm/ako-operator/values.yaml]: Removed.

File: ako-operator/helm/ako-operator/values.yaml (hunk 1 above)
Comment (hemantavi) [reply ako-operator/helm/ako-operator/values.yaml]: Deleted.

**Review by monotosh-avi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**